Title: socket.recv() Method
URL: https://docs.python.org/3/library/socket.html
Summary: The socket.recv() method receives data from a socket, returning a bytes object up to a specified buffer size. An empty return value indicates disconnection. The method behavior has changed in version 3.5 to retry interrupted system calls instead of raising an InterruptedError.
---

socket.recv(_bufsize_[, _flags_])[¶](https://docs.python.org/3/library/socket.html#socket.socket.recv "Link to this definition")
    
Receive data from the socket. The return value is a bytes object representing the data received. The maximum amount of data to be received at once is specified by _bufsize_. A returned empty bytes object indicates that the client has disconnected. See the Unix manual page _[recv(2)](https://manpages.debian.org/recv\(2\))_ for the meaning of the optional argument _flags_ ; it defaults to zero.
Changed in version 3.5: If the system call is interrupted and the signal handler does not raise an exception, the method now retries the system call instead of raising an [`InterruptedError`](https://docs.python.org/3/library/exceptions.html#InterruptedError "InterruptedError") exception (see [**PEP 475**](https://peps.python.org/pep-0475/) for the rationale). 

socket.recvfrom(_bufsize_[, _flags_])[¶](https://docs.python.org/3/library/socket.html#socket.socket.recvfrom "Link to this definition")
    
Receive data from the socket. The return value is a pair `(bytes, address)` where _bytes_ is a bytes object representing the data received and _address_ is the address of the socket sending the data. See the Unix manual page _[recv(2)](https://manpages.debian.org/recv\(2\))_ for the meaning of the optional argument _flags_ ; it defaults to zero. (The format of _address_ depends on the address family — see above.)
Changed in version 3.5: If the system call is interrupted and the signal handler does not raise an exception, the method now retries the system call instead of raising an [`InterruptedError`](https://docs.python.org/3/library/exceptions.html#InterruptedError "InterruptedError") exception (see [**PEP 475**](https://peps.python.org/pep-0475/) for the rationale).
Changed in version 3.7: For multicast IPv6 address, first item of _address_ does not contain `%scope_id` part anymore. In order to get full IPv6 address use [`getnameinfo()`](https://docs.python.org/3/library/socket.html#socket.getnameinfo "socket.getnameinfo").